src/sst/core/params.cc

```diff
@@ -242,6 +242,140 @@ Params::getKey(const std::string& str)
 }
 
 
+/**
+   Private function for parsing string into array tokens.
+
+ * Parses a string representing an array of tokens.  It is tailored to
+ * the types strings you get when passing a python list as the param
+ * string.  When you call addParam() on a python list in the input
+ * file, it will call the str() function on the list, which creates a
+ * string with the following format: [item1, item2, item3]
+
+ * The format of the items depends on where they came from.  The
+ * string for the items are generated by calling the repr() function
+ * on them.  For strings, this means they will typically be enclosed
+ * in single quotes.  It is possible that they end up enclosed in
+ * double quotes if the string itself contains a single quote.  For
+ * strings which contain both single and double quotes, the repr()
+ * will create a single quoted string with all internal single quotes
+ * escaped with '\'.  Most other items used in SST do not enclose the
+ * string in quotes, though any string that contains a comma would
+ * need to be enclosed in quotes, since the comma is the delimiter
+ * character used.  This is not done automatically, so if you have
+ * something that generates a commma in the string created by repr(),
+ * you may need to create an array string manually.  Also, any string
+ * that starts with a quote character, must end with the same quote
+ * character.
+
+ * The string passed into this function should have the [] already
+ * removed from the string.  Tokens are generated by splitting the
+ * string on commas that are not within quotes (double or single).
+ * All whitespace at the beginning and end of a token is ignored
+ * (unless inside quotes).  Once the tokens are generated, any quoted
+ * string will have the front and back quotes removed.  The '/' for
```

Comment:
Should the '/' be '\' ?

Reply:
Fixed here and in params.h.  Also added some examples of using double and single quotes in strings and how they show up after parsing.